nixos/modules/programs/rust-motd.nix

```diff
@@ -27,10 +55,43 @@ in {
         For possible formats, please refer to {manpage}`systemd.time(7)`.
       '';
     };
+    order = mkOption {
+      type = types.listOf types.str;
+      default = attrNames cfg.settings;
+      defaultText = literalExpression "attrNames cfg.settings";
+      description = mdDoc ''
+        The order of the sections in [](#opt-programs.rust-motd.settings) implies
+        the order of sections in the motd. Since attribute sets in Nix are always
+        ordered alphabetically internally this means that
+
+        ```nix
+        {
+          uptime = { /* ... */ };
+          banner = { /* ... */ };
+        }
+        ```
+
+        will still have `banner` displayed before `uptime`.
+
+        To work around that, this option can be used to define the order of all keys,
+        i.e.
+
+        ```nix
+        {
+          order = [
+            "uptime"
+            "banner"
+          ];
+        }
+        ```
+
+        makes sure that `uptime` is placed before `banner` in the motd.
+      '';
+    };
     settings = mkOption {
-      type = types.submodule {
+      type = types.attrsOf (types.submodule {
         freeformType = format.type;
-      };
+      });
```

Comment:
I don't think this change is necessary anymore.

Reply:
We need an attribute-set at the top-level and `formts.toml.type` accepts strings and ints on the topmost level as well, so I don't think so.

Reply:
Fair! Btw `freeformType` isn't needed unless there's actual options declared. So in this case `type = types.attrsOf format.type` would be enough. Probably also more efficient.

Reply:
Oof yeah, I don't even remember why I added it in the first place when writing that module. Removed.